packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,917 @@
+            activeElement instanceof FoundationElement
+            && !(activeElement instanceof TableRow)
+            && !(activeElement instanceof TableCell)
+            && !(activeElement instanceof TableCellView)
+        ) {
+            break;
+        }
+    }
+
+    return activeElement as HTMLElement;
+}
+
+describe('Table keyboard navigation', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+
+    async function addActionMenu(column: TableColumn): Promise<void> {
+        column.actionMenuSlot = `${column.id}-action-menu-slot`;
+        column.actionMenuLabel = 'Actions';
+
+        const menu = document.createElement('nimble-menu');
+        menu.slot = column.actionMenuSlot;
+
+        const menuItem1 = document.createElement('nimble-menu-item');
+        menuItem1.textContent = 'menu item 1';
+        menu.appendChild(menuItem1);
+
+        element.appendChild(menu);
+
+        await waitForUpdatesAsync();
+    }
+
+    function isCellActionMenuVisible(
+        rowIndex: number,
+        columnIndex: number
+    ): boolean {
+        const cell = pageObject.getCell(rowIndex, columnIndex);
+        if (cell.actionMenuButton) {
+            return cell.actionMenuButton.getBoundingClientRect().width > 0;
+        }
+        throw new Error('Cell action menu button not found');
+    }
+
+    async function verifyLastTabKeyEventBehavior(
+        keyEvent: KeyboardEvent
+    ): Promise<void> {
+        // Note: Dispatching key events with dispatchEvent is not sufficient to trigger the browser's built-in Tab behavior.
+        // From these tests, we can only check defaultPrevented (and assume that defaultPrevented=false means that the browser will
+        // focus the expected element outside the table).
+        expect(keyEvent.defaultPrevented).toBe(false);
+        // Currently KeyboardNavigationManager will set tabindex=-1 on the focused element, but not blur it, relying on the browser's
+        // built in behavior for that (for correct behavior in Safari).
+        const focusedElement = currentFocusedElement()!;
+        expect(focusedElement).not.toBeNull();
+        expect(focusedElement.tabIndex).toBe(-1);
+        // Manually blur so we can ensure that the table doesn't re-take focus
+        focusedElement.blur();
+        await waitForUpdatesAsync();
+        expect(element.shadowRoot?.activeElement).toBeNull();
+    }
+
+    async function sendKeyPress(
+        target: HTMLElement,
+        key: string,
+        init?: KeyboardEventInit
+    ): Promise<KeyboardEvent> {
+        const event = new KeyboardEvent('keydown', {
+            key,
+            cancelable: true,
+            bubbles: true,
+            ...init
+        });
+        target.dispatchEvent(event);
+        await waitForUpdatesAsync();
+        return event;
+    }
+
+    async function sendKeyPressToTable(
```

Comment:
Do you need this if you always send the event to `currentFocusedElement` (via `sendKeyPress`), since the event is set to `bubble`?

Reply:
For some reason, it doesn't look like sending key presses to `currentFocusedElement()` triggers the table key listeners (either the capture-phase one or the bubbling-phase one).